### DISCUSSION TITLE
ORDERED morphological model with user-supplied rates and state freqs

### DISCUSSION THREAD

**Discussion by AllenWLynch** [iqtree/iqtree2#201] [Q&A]
Hi, I am working with ordered morphological data for which I have known substitution rates. I did not see in the documentation if I can provide these rates to the ordered model similar to GTR{a,b,c}, or if there is an API similar to the user-supplied protein and DNA models. There was a similar question asked on the old google forum (https://groups.google.com/g/iqtree/c/OnW4kDcJxTM/m/ZgqSRyA_CgAJ, 2017).

Has this been implemented? If not, would my best option be to make my own subclass of `ModelMorphology` with custom rates?

Any guidance is appreciated. Many thanks,
AL

**Comment by bqminh** [comment]
From the code:

https://github.com/iqtree/iqtree2/blob/c8be3c3117a0511a180c83a0d1768c0b3b520f41/model/modelmorphology.cpp#L37

It looks like yes, IQ-TREE can read custom rates from a file, just like for DNA/AA/codon models.

you can try something like:

```
iqtree2 -s MORPH_ALN -m CUSTOM_RATES_FILE+G
```

A `git blame` tells that I introduced this code with commit 9ff43a1 from 2017-11-30. This commit also tells me that you can try `-m GTRX{a,b,c,...}`. Let me know if it works.
